Realm/RLMProperty.mm

```diff
@@ -354,42 +373,40 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
 
     // convert array types to objc variant
     if ([rawType isEqualToString:@"@\"RLMArray\""]) {
-        rawType = [NSString stringWithFormat:@"@\"RLMArray<%@>\"", [propertyValue objectClassName]];
+        RLMArray *value = propertyValue;
+        _type = RLMPropertyTypeArray;
+        _objectClassName = value.objectClassName;
+        if (_objectClassName) {
+            // FIXME: validate
```

Comment:
Is this an existing FIXME that was moved around, or something new that should be addressed?